tests/phpunit/unit-tests/traits/llms-test-trait-singleton.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Tests for {@see LLMS_Trait_Singleton}.
+ *
+ * @group Traits
+ *
+ * @since [version]
+ */
+class LLMS_Test_Singleton_Trait extends LLMS_UnitTestCase {
+
+	/**
+	 * Dynamic class name of the mock class.
+	 *
+	 * Even though this property contains a string, it is documented as a class so that it can be used like this:
+	 * `$this->mock_class::instance()`
+	 *
+	 * @since [version]
+	 *
+	 * @var LLMS_Trait_Singleton|object
+	 */
+	protected $mock_class;
+
+	/**
+	 * Setup before running each test in this class.
+	 *
+	 * @since [version]
+	 * @noinspection PhpHierarchyChecksInspection
+	 */
+	public function setUp() {
+
+		parent::setUp();
+
+		# Instantiate an anonymous class that uses the trait to be tested.
+		$mock = new class {
+
+			use LLMS_Trait_Singleton;
+
+			protected $color;
+
+			protected static $_instance = null;
+
+			public static function deprecated_instance() {
+				if ( is_null( self::$_instance ) ) {
+					self::$_instance = new self();
+				}
+
+				return self::$_instance;
+			}
+
+			public static function init() {
+				self::$_instance = null;
+				self::$instance  = null;
+			}
+
+			public function get_color() {
+				return $this->color;
+			}
+
+			public function set_color( $color ) {
+				$this->color = $color;
+			}
+		};
+
+		$this->mock_class = get_class( $mock );
```

Comment:
I think this will fix tests failing on php8

```suggestion
		$this->mock_class = new class {

			use LLMS_Trait_Singleton;

			protected $color;

			protected static $_instance = null;

			public static function deprecated_instance() {
				if ( is_null( self::$_instance ) ) {
					self::$_instance = new self();
				}

				return self::$_instance;
			}

			public static function init() {
				self::$_instance = null;
				self::$instance  = null;
			}

			public function get_color() {
				return $this->color;
			}

			public function set_color( $color ) {
				$this->color = $color;
			}
		};
```

Reply:
I originally wrote it this way, storing an object. But then I realized that it wasn't a true test of static properties.

I'm looking into what's up with PHP 8 and the Reflection classes.

Reply:
[`ReflectionProperty::getValue()`](https://www.php.net/manual/en/reflectionproperty.getvalue.php), used by [`LLMS_Unit_Test_Util ::get_private_property_value()`](https://github.com/gocodebox/lifterlms-tests/blob/trunk/framework/class-llms-unit-test-util.php#L93), requires an object for non-static properties and `null` for static properties.

Options:
1. Update `LLMS_Unit_Test_Util ::get_private_property_value()` to check if property is static.
2. Add a new method to handle private static properties. `LLMS_Unit_Test_Util ::get_private_static_property_value()` 
3. Use [`ReflectionClass::getDefaultProperties()`](https://www.php.net/manual/en/reflectionclass.getdefaultproperties.php) instead of `ReflectionProperty::getValue()`. **Needs testing.**

Reply:
1. sounds good

Reply:
After some more reading, I agree that #1 is the best option.

Reply:
Of course, this means that now this PR is dependent on a lifterlms-tests PR.

Reply:
That's okay, it happens

Reply:
I made a lifterlms-tests PR https://github.com/gocodebox/lifterlms-tests/pull/9.